homeassistant/components/dynalite/config_flow.py

```diff
@@ -18,19 +20,69 @@ def __init__(self) -> None:
         """Initialize the Dynalite flow."""
         self.host = None
 
+    def existing_host_entry(self, config) -> Any:
+        """Check if a specific host is already configured and return its entry or None."""
+        for entry in self.hass.config_entries.async_entries(DOMAIN):
+            if entry.data[CONF_HOST] == config[CONF_HOST]:
+                return entry
+        return None
+
+    async def check_bridge_connection(self, config) -> bool:
+        """Check whether this bridge is available to connect."""
+        bridge = DynaliteBridge(self.hass, config)
+        if not await bridge.async_setup():
+            LOGGER.error("Unable to setup bridge - config=%s", config)
+            return False
+        return True
+
     async def async_step_import(self, import_info: Dict[str, Any]) -> Any:
         """Import a new bridge as a config entry."""
         LOGGER.debug("Starting async_step_import - %s", import_info)
         host = import_info[CONF_HOST]
-        for entry in self.hass.config_entries.async_entries(DOMAIN):
-            if entry.data[CONF_HOST] == host:
-                if entry.data != import_info:
-                    self.hass.config_entries.async_update_entry(entry, data=import_info)
-                return self.async_abort(reason="already_configured")
+        entry = self.existing_host_entry(import_info)
+        if entry:
+            if entry.data != import_info:
+                self.hass.config_entries.async_update_entry(entry, data=import_info)
+            return self.async_abort(reason="already_configured")
         # New entry
-        bridge = DynaliteBridge(self.hass, import_info)
-        if not await bridge.async_setup():
+        if not await self.check_bridge_connection(import_info):
             LOGGER.error("Unable to setup bridge - import info=%s", import_info)
-            return self.async_abort(reason="no_connection")
+            return self.async_abort(reason="cannot_connect")
         LOGGER.debug("Creating entry for the bridge - %s", import_info)
         return self.async_create_entry(title=host, data=import_info)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        # This is for backwards compatibility.
+        return await self.async_step_init(user_input)
+
+    async def async_step_init(self, user_input=None):
+        """Handle a flow start."""
+
+        def form_schema(defaults):
+            return vol.Schema(
+                {
+                    vol.Optional(CONF_NAME, default=defaults.get(CONF_NAME)): str,
+                    vol.Required(CONF_HOST, default=defaults.get(CONF_HOST)): str,
+                    vol.Optional(CONF_PORT, default=defaults.get(CONF_PORT)): int,
+                }
+            )
```

Comment:
Let's move this to the class instead of the method, so it's easier to follow. You can also decorate it with `@callback` to show that it's safe to be called in an async context (you'll need to import `callback` (from `core`, IIRC).
I think you can also show the form, from this, not just create the schema (have an optional `errors` argument).

Reply:
took another stab at it. please let me know if this is what you meant